recipes/boost/all/conanfile.py

```diff
@@ -247,6 +263,28 @@ def config_options(self):
             # Shared builds of numa do not link on Visual Studio due to missing symbols
             self.options.numa = False
 
+        if tools.Version(self.version) >= "1.76.0":
+            # Starting from 1.76.0, Boost.Math requires a c++11 capable compiler
+            # ==> disable it by default for older compilers or c++ standards
+
+            def disable_math():
+                super_modules = self._all_super_modules("math")
+                for smod in super_modules:
+                    try:
+                        setattr(self.options, "without_{}".format(smod), True)
+                    except ConanException:
+                        pass
+
+            if self.settings.compiler.cppstd:
+                if not tools.valid_min_cppstd(self, 11):
+                    disable_math()
+            else:
+                min_compiler_version = self._min_compiler_version_default_cxx11
+                if min_compiler_version is None:
+                    self.output.warn("Assuming the compiler supports c++11 by default")
+                elif tools.Version(self.settings.compiler.version) < min_compiler_version:
+                    disable_math()
```

Comment:
Why not inject C++11 flag instead if math is enabled?

Reply:
I don't think it's right thing to do. if compiler.cppstd is set explicitly, it should be passed according to the setting. if it's not defined at all, it means just follow the compiler's default, whatever it is. in that case, injecting standard different from the default will only cause confusion, or even problems (as boost changes ABI for C++11 which might be undesired).

Reply:
this is what we do indirectly in almost all libs based on CMake, since usually they force CXX_STANDARD to the minimum standard required to properly build.

Reply:
I hope that this won't cause an influx of issues reporting missing boost math libraries.

Reply:
yes, we're doing it now, but it doesn't mean we're doing the right thing :)
I think the plan is to eventually add `cppstd` support for CCI and process it accordingly, instead of silently injecting `-std=c++xx` in some recipes.

Reply:
We don't inject this flag ourself, it's already there usually in underlying build files of libraries, this will required too much patches I think.

Reply:
for boost, the flag is controlled by `b2 cxxstd=<version>` and it's not hard-coded. so for us, it should be a simple mapping `compiler.cppstd` <-> `cxxstd`.
for other builds systems, it will be another option (e.g. `CMAKE_CXX_STANDARD` in CMake or `cpp_std` in meson), thus similar mapping will happen in the recipe or build helper.
some projects tend to hard code specific C++ standard in their makefiles (e.g. hardcoding `CXXFLAGS=-std=c++<version>` or `set(CMAKE_CXX_STANDARD <version>`). should we do something about it?
there are two ways:
- provide patches to remove hardcoded C++ standard flags
- assume the hard-coded C++ standard is the only one supported by the library, thus, throw `ConanInvalidConfigurations` for all other `compiler.cppstd` values besides hard-coded one

I personally think the second approach is the most correct one.
that we're doing right now is a complete mess, and the only excuse is that CCI doesn't currently support `compiler.cppstd`, and if we try to do things correctly, we will be unable to build many libraries. 
but so far it works fine, and we had almost no reports of it caused any actual problems.
doing things correctly, on other hand, will increase pressure on maintainers, as you pointed out, as well it will make CI more complicated and will require way more resources (and will make builds even slower, multiply boost build time by the number of active C++ standards). 
probably, the benefit doesn't worth the effort right now, and there are other areas which will bring more benefit for much less effort. still, in future, some C++ standard may introduce a huge ABI break, which will make "correct" handling of C++ standard in CCI a vital and must have feature.

Reply:
> I personally think the second approach is the most correct one.

This is the policy we have, before CMake was the main build script it was a wild west and it's the on sensible way...

As a consumer if you build locally you expect the project to be configured as intended. It's not our place to change that value... nor test all possible combinations and patch in between cppstd